### PR TITLE
Align the list of HTTP reasons with the official one.

### PR DESCRIPTION
**Pull request by carloseltuerto**
This follows the conclusion of [Issue 1543](https://github.com/envoyproxy/envoy-mobile/issues/1543)

Description: For HTTP reasons, use this official list: https://www.iana.org/assignments/http-status-codes/http-status-codes.xhtml
Risk Level: none
Testing: tests still pass - cosmetic change
Docs Changes: N/A
Release Notes: N/A
Signed-off-by: Charles Le Borgne <cleborgne@google.com>

### CODE REVIEW

**Review by buildbreaker — APPROVED** [review]
Nice!

**Review by DavidSchinazi — DISMISSED** [review]

File: test/java/org/chromium/net/CronetUrlRequestTest.java [hunk 1]
```diff
@@ -513,7 +516,7 @@ public void testAddHeader() throws Exception {
   @Test
   @SmallTest
   @Feature({"Cronet"})
-  @Ignore("MockWebServer does not handle header field with multiple values like NativeTestServer")
+  @Ignore("Debug me")
```
Comment (buildbreaker) [inline test/java/org/chromium/net/CronetUrlRequestTest.java:519]: Is this broken?

**Review by buildbreaker — APPROVED** [review]

File: test/java/org/chromium/net/CronetUrlRequestTest.java (hunk 1 above)
Comment (carloseltuerto) [reply test/java/org/chromium/net/CronetUrlRequestTest.java:519]: There are two tests for which I don't have a good explanation yet. I will take care of them later.

Comment (buildbreaker) [reply test/java/org/chromium/net/CronetUrlRequestTest.java:519]: Sounds good!

**Pull request merged**